### PR TITLE
Fixed flow issues in flux-utils-todomvc example

### PR DESCRIPTION
**Pull request by eikeon**
Here's my first pass at fixing the flow issues in the flux-utils-todomvc example #285.

With these changes I am no longer getting any flow errors running against flux from master. There are a number of fixes such as #317 since the last release.

### CODE REVIEW

File: examples/flux-utils-todomvc/js/components/TodoItem.react.js [hunk 1]
```diff
@@ -27,12 +27,12 @@ type State = {
   isEditing: boolean,
 };
 
-export default class TodoItem extends Component<{}, Props, State> {
-  state = {
+export default class TodoItem extends Component<void, Props, State> {
+  state:State = {
```
Comment (kyldvs) [inline examples/flux-utils-todomvc/js/components/TodoItem.react.js]: <nit> add a space, `state: State..`

File: examples/flux-utils-todomvc/js/components/TodoItem.react.js [hunk 2]
```diff
@@ -67,9 +67,9 @@ export default class TodoItem extends Component<{}, Props, State> {
         {input}
       </li>
     );
-  }
+  };
 
-  _onToggleComplete = () => {
+  _onToggleComplete() : void {
```
Comment (kyldvs) [inline examples/flux-utils-todomvc/js/components/TodoItem.react.js]: <nit> remove the extra space, `_onToggleComplete(): void`

File: examples/flux-utils-todomvc/js/components/TodoItem.react.js [hunk 3]
```diff
@@ -82,13 +82,13 @@ export default class TodoItem extends Component<{}, Props, State> {
         id: todo.id,
       });
     }
-  }
+  };
 
-  _onDoubleClick = () => {
+  _onDoubleClick() : void {
```
Comment (kyldvs) [inline examples/flux-utils-todomvc/js/components/TodoItem.react.js]: extra space

File: examples/flux-utils-todomvc/js/components/TodoItem.react.js [hunk 4]
```diff
@@ -82,13 +82,13 @@ export default class TodoItem extends Component<{}, Props, State> {
...
     this.setState({isEditing: true});
-  }
+  };
 
-  _onSave = (text) => {
+  _onSave(text: string) : void {
```
Comment (kyldvs) [inline examples/flux-utils-todomvc/js/components/TodoItem.react.js]: extra space

File: examples/flux-utils-todomvc/js/components/TodoItem.react.js [hunk 5]
```diff
@@ -96,13 +96,13 @@ export default class TodoItem extends Component<{}, Props, State> {
       text,
     });
     this.setState({isEditing: false});
-  }
+  };
 
-  _onDestroyClick = () => {
+  _onDestroyClick() : void {
```
Comment (kyldvs) [inline examples/flux-utils-todomvc/js/components/TodoItem.react.js]: extra space

File: examples/flux-utils-todomvc/js/components/TodoTextInput.react.js [hunk 6]
```diff
@@ -28,12 +28,12 @@ type State = {
   value: string,
 };
 
-export default class TodoTextInput extends Component<{}, Props, State> {
-  state = {
+export default class TodoTextInput extends Component<void, Props, State> {
+  state : State = {
```
Comment (kyldvs) [inline examples/flux-utils-todomvc/js/components/TodoTextInput.react.js]: extra space

File: examples/flux-utils-todomvc/js/components/TodoItem.react.js [hunk 7]
```diff
@@ -67,9 +67,9 @@ export default class TodoItem extends Component<{}, Props, State> {
         {input}
       </li>
     );
-  }
+  };
```
Comment (kyldvs) [inline examples/flux-utils-todomvc/js/components/TodoItem.react.js]: no semicolons after methods in a class

**Comment by kyldvs** [comment]
Looks great, happy to merge this if you can clean up the nits. Flow type annotations should always be `foo: bar`, not `foo:bar` or `foo : bar`. And the semicolons after class method are not necessary, after property declarations is fine.

**Comment by eikeon** [comment]
Will do. I'm also noticing some other issues

**Comment by eikeon** [comment]
@kyldvs I've cleaned up the commit and amended it. Flow is now reporting no errors for me: 
```
flux-utils-todomvc$ flow
No errors!
```

But I'm still not sure what the .flowconfig should look like? Not sure it's picking up on any React declarations.

**Comment by kyldvs** [comment]
Awesome. I think the `.flowconfig` is probably fine. I believe react definitions are built into flow: https://github.com/facebook/flow/blob/master/lib/react.js

Thanks for doing this!

**Comment by eikeon** [comment]
You're welcome. And thank you!

**Pull request merged**